libs/convert/utils.go

```diff
@@ -0,0 +1,84 @@
+package convert
+
+import (
+	"reflect"
+	"slices"
+	"strings"
+	"unicode"
+
+	"github.com/databricks/cli/libs/config"
+	"github.com/databricks/cli/libs/config/convert"
+)
+
+var skipFields = []string{"ForceSendFields", "Format"}
+
+// Returns config name to be used in YAML configuration for
+// the config value passed. Uses the name defined in 'json' tag
+// for the structure
+func key(v any, name string) string {
+	t := reflect.TypeOf(v)
+	if t.Kind() == reflect.Pointer {
+		t = reflect.TypeOf(v).Elem()
+	}
+	field, ok := t.FieldByName(name)
+	if !ok {
+		return name
+	}
+	key, _, _ := strings.Cut(field.Tag.Get("json"), ",")
+	if key == "-" || key == "" {
+		return name
+	}
```

Comment:
We can ignore fields without a `json` tag or where it is set to `-`.

Then we also don't need to explicitly ignore `ForceSendFields`, as it is not part of the `json` representation.